components/mpas-ocean/src/mode_forward/mpas_ocn_forward_mode.F

```diff
@@ -821,7 +831,17 @@ function ocn_forward_mode_run(domain) result(ierr)!{{{
          endif
 
          ! read in next time level data required for time-varying forcing
-         call ocn_time_varying_forcing_get(domain % streamManager, domain, domain % clock)
+         if (timeIntegratorChoice /= 4) then
+           ! if not using RK4, calculate time varying forcing terms once per
+           ! time-step as opposed at each RK substage as implemented in RK4
+           call ocn_time_varying_forcing_get(domain % streamManager, domain, domain % clock)
+         else
+           if (config_use_time_varying_atmospheric_forcing .or. &
+               config_use_time_varying_land_ice_forcing) then
+             ! increment forcing clock to next time-step
+             call mpas_advance_forcing_clock(forcingGroupHead, dt)
```

Comment:
Actually, maybe instead of setting the `forcingTimeIncrementRK4` to zero in terms of having a light coupling we should not advance the forcing clock here. Please check if this statement is true. Thanks!